assets/js/modules/adsense/components/module/ModuleOverviewWidget/index.js

```diff
@@ -0,0 +1,142 @@
+import { STORE_NAME } from '../../../datastore/constants';
+import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
+import { isZeroReport } from '../../../util';
+import ProgressBar from '../../../../../components/ProgressBar';
+import Header from './Header';
+import Overview from './Overview';
+import Stats from './Stats';
+import Data from 'googlesitekit-data';
+const { useSelect } = Data;
+
+const ModuleOverviewWidget = ( { Widget, WidgetReportZero, WidgetReportError } ) => {
+	const metrics = {
+		EARNINGS: __( 'Earnings', 'google-site-kit' ),
+		PAGE_VIEWS_RPM: __( 'Page RPM', 'google-site-kit' ),
+		IMPRESSIONS: __( 'Impressions', 'google-site-kit' ),
+		PAGE_VIEWS_CTR: __( 'Page CTR', 'google-site-kit' ),
+	};
+	const [ selectedStats, setSelectedStats ] = useState( 0 );
+
+	const {
+		startDate,
+		endDate,
+		compareStartDate,
+		compareEndDate,
+	} = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( { compare: true } ) );
+
+	const currentRangeArgs = {
+		metrics: Object.keys( metrics ),
+		startDate,
+		endDate,
+	};
+	const previousRangeArgs = {
+		metrics: Object.keys( metrics ),
+		startDate: compareStartDate,
+		endDate: compareEndDate,
+	};
+	const currentRangeChartArgs = {
+		...currentRangeArgs,
+		dimensions: [ 'DATE' ],
+	};
+	const previousRangeChartArgs = {
+		...previousRangeArgs,
+		dimensions: [ 'DATE' ],
+	};
+
+	const currentRangeData = useSelect( ( select ) => select( STORE_NAME ).getReport( currentRangeArgs ) );
+	const previousRangeData = useSelect( ( select ) => select( STORE_NAME ).getReport( previousRangeArgs ) );
+	const currentRangeChartData = useSelect( ( select ) => select( STORE_NAME ).getReport( currentRangeChartArgs ) );
+	const previousRangeChartData = useSelect( ( select ) => select( STORE_NAME ).getReport( previousRangeChartArgs ) );
+
+	const currentRangeLoading = useSelect( ( select ) => ! select( STORE_NAME ).hasFinishedResolution( 'getReport', [ currentRangeArgs ] ) );
+	const previousRangeLoading = useSelect( ( select ) => ! select( STORE_NAME ).hasFinishedResolution( 'getReport', [ previousRangeArgs ] ) );
+	const currentRangeChartLoading = useSelect( ( select ) => ! select( STORE_NAME ).hasFinishedResolution( 'getReport', [ currentRangeChartArgs ] ) );
+	const previousRangeChartLoading = useSelect( ( select ) => ! select( STORE_NAME ).hasFinishedResolution( 'getReport', [ previousRangeChartArgs ] ) );
+
+	const currentRangeError = useSelect( ( select ) => select( STORE_NAME ).getErrorForSelector( 'getReport', [ currentRangeArgs ] ) );
+	const previousRangeError = useSelect( ( select ) => select( STORE_NAME ).getErrorForSelector( 'getReport', [ previousRangeArgs ] ) );
+	const currentRangeChartError = useSelect( ( select ) => select( STORE_NAME ).getErrorForSelector( 'getReport', [ currentRangeChartArgs ] ) );
+	const previousRangeChartError = useSelect( ( select ) => select( STORE_NAME ).getErrorForSelector( 'getReport', [ previousRangeChartArgs ] ) );
```

Comment:
These could be combined into 2 selects. We generally do a single `useSelect` with a single select within, but the convention is to use a single `useSelect` for each value that we need in the component scope. In this case, we only care about `loading` and `error` – each individual value isn't important in this case, we just want the first one or none. Furthermore, this is a little more efficient because within the selector function we can return as soon as we've found something rather than selecting them all unconditionally.

Reply:
@aaemnnosttv I made the updates, but not sure if it's what you meant. Can you check?